docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,629 @@
+
+The local copy of your repository needs to track both your remote repository and the original repository from which it was forked. By default, when cloning, a link is created back to your copy but not the original repository.
+
+Let's find out what remotes currently exist.
+
+Type
+
+```
+git remote
+```
+
+![submit_pr_remote_tracking_1.png](./imgs/submit_pr_remote_tracking_1.png)
+
+There should be one remote named **origin** and it points to the forked remote repository.
+
+If a remote named **upstream** (pointing to the original remote repository) does not exist, you need to create it.
+
+In your browser, navigate to original repository  <https://github.com/strongloop/loopback-next> and ensure that the `master` branch is selected.
+
+
+![submit_pr_remote_tracking_2.png](./imgs/submit_pr_remote_tracking_2.png)
+
+Click on the `Clone or download` button, leave the cloning option as `Clone with SSH` and press the `copy to clipboard` icon.
+
+![submit_pr_remote_tracking_3.png](./imgs/submit_pr_remote_tracking_3.png)
+
+
+Return to your terminal window and type :
+
+```
+git remote add upstream git@github.com:strongloop/loopback-next.git
+```
+
+then type :
+
+```
+git remote
+```
+
+there should now be two remotes : **origin** and **upstream**.
+
+![submit_pr_remote_tracking_4.png](./imgs/submit_pr_remote_tracking_4.png)
+
+Typing
+
+```
+git remote -v
+```
+
+provides more details.
+
+![submit_pr_remote_tracking_5.png](./imgs/submit_pr_remote_tracking_5.png)
+
+### 8. Install dependencies
+
+The `loopback-next` has several package dependencies.
+
+Type `npm install` to install them.
+
+![submit_pr_npm_install_1.png](./imgs/submit_pr_npm_install_1.png)
+
+
+### 9. Do some work in your local feature branch directory
+
+Whether you are contributing to code or documentation, be sure to make all your changes inside in the local feature branch directory.
+
+Be sure to read, understand, and abide by the instructions provided in [Contributing code](https://loopback.io/doc/en/contrib/code-contrib.html), [Contributing to docs](https://loopback.io/doc/en/contrib/doc-contrib.html), and
+[Contributing to LoopBack](https://github.com/strongloop/loopback-next/blob/master/docs/CONTRIBUTING.md)
+
+### 10. Verifying your code changes
+
+Before `staging your changes` (the next step), be sure your files are free of formatting, syntax, and logical/execution errors.
+
+To flush out any **formatting** issues in code, run `npm run lint`. If there are issues, run `npm run lint:fix`.
+
+To flush out any **syntax** issues in code, run `npm run build`. If there are issues, fix them, and run `npm run build` again. Repeat until all issues are gone.
+
+To flush out any **logical/execution** issues in code, run `npm run test`. This will run the existing mocha test cases, and any new test cases you have added. If there are issues, fix them, run `npm build`, and run `npm run test` again. Repeat until all issues are gone.
+
+### 11. Verifying your documentation changes
```

Comment:
We can probably leave it to CI.

Reply:
I liked your earlier statement:
"
Maybe the simplest approach is `npm run lint:fix && npm test`.
"
Also, doesn't CI simply run the same thing that a user runs on his laptop? If so, it would save them time to run on the laptop first, to flush out all problems locally, before submitting the PR.